org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/junit/Junit4Fragment2.xtend

```diff
@@ -27,11 +27,19 @@ class Junit4Fragment2 extends AbstractStubGeneratingFragment {
 			"org.eclipse.xtext.testing"
 	}
 	
+	def protected getXbaseTestingPackage() {
+		if (useDeprecatedClasses)
+			"org.eclipse.xtext.xbase.junit"
+		else
+			"org.eclipse.xtext.xbase.testing"
+	}
+	
 	override generate() {
 		if (projectConfig.runtimeTest.manifest !== null) {
 			projectConfig.runtimeTest.manifest => [
 				requiredBundles.addAll(
 					testingPackage,
+					xbaseTestingPackage,
```

Comment:
I see. Actually some classes in this bundle, mainly CompilationTestHelper, are valuable also in projects that do not use Xbase, just for testing the code generator (at least, that's the path I follow also in my book, for testing generated code of a DSL not based on Xbase).

@cdietrich was proposing to add a flag to skip the generation of this dependency, https://github.com/eclipse/xtext-core/issues/230#issuecomment-271631841, and, at least for me, I would prefer this flag in the fragment. Something like skipXbaseTestingPackage (default to false).

Summarizing, I'd like to have this dependency by default :) but then I leave the final decision to you.